implementation/src/main/java/io/smallrye/graphql/bootstrap/schema/helper/CollectionHelper.java

```diff
@@ -40,20 +38,17 @@
      * @param type the collection class
      * @return the collection
      */
-    public Collection newCollection(Class type) {
-        if (type.equals(Collection.class) || type.equals(List.class)) {
-            return new ArrayList();
-        } else if (type.equals(Set.class)) {
-            return new HashSet();
-        } else {
-            try {
-                Constructor constructor = type.getConstructor();
-                constructor.setAccessible(true);
-                return (Collection) constructor.newInstance();
-            } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException ex) {
-                LOG.error("Can not create new collection of [" + type.getName() + "]", ex);
-                return new ArrayList(); // default ?
+    public static Collection<?> newCollection(Class<?> type) {
+        try {
+            return (Collection<?>) type.newInstance();
+        } catch (Exception ex) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Can not create no-arg collection instance of [" + type + "]", ex);
```

Comment:
this will log `... instance of [class ...]` while `getName()` would be sufficient.

Reply:
Sure, I'll change it - since it's a public method, it's possible (although unlikely) that somebody could pass a null into the `newCollection` method, so changing it to print the `getName()` method could introduce a NullPointer.  I'll change it to `(type == null ? "null" : type.getName())`.